packages/integration-test/test/layers.test.ts

```diff
@@ -424,7 +424,8 @@ describe('Request system', () => {
     const request1CancelHash: RequestLogicTypes.ICancelParameters = {
       requestId: requestId1,
     };
-    await requestLogic.cancelRequest(request1CancelHash, payeeIdentity);
+    const cancel = await requestLogic.cancelRequest(request1CancelHash, payeeIdentity);
+    await new Promise((resolve) => cancel.on('confirmed', resolve));
```

Comment:
about this and all await on "confirmed": i'm almost sure this is because we no longer do a long query to get the metadata, so the request is no longer confirmed straight away

Reply:
Nice! We should replace all those too I think, to stabilize our tests. I'll open a PR later on for this.
```ts
    // wait a bit
    // eslint-disable-next-line no-magic-numbers
    await new Promise((r: any): any => setTimeout(r, 1000));
```

Reply:
I did a bunch already, but there's a lot to do. We should maybe introduce a util for that.

Reply:
Yes I saw that, very nice! I agree for the util.